internal/provider/provider.go

```diff
@@ -235,6 +235,10 @@ func (p *infisicalProvider) Configure(ctx context.Context, req provider.Configur
 		clientSecret = config.ClientSecret.ValueString()
 	}
 
+	if config.Auth != nil && !config.Auth.Token.IsNull() {
+		token = config.Auth.Token.ValueString()
+	}
+
```

Comment:
doesn't this do the same thing as the following below? 

<img width="666" height="105" alt="image" src="https://github.com/user-attachments/assets/a15a7a4b-0d33-4040-bde0-5ea9f5daa3bb" />